### DISCUSSION TITLE
Telegram - rule trigger

### DISCUSSION THREAD

**Discussion by mantielero** [arendst/Tasmota#15418] [Q&A] [answered]
I am trying to do something like this: https://www.youtube.com/watch?v=TOxzQSdivVI

And maybe I am not using well Telegram with Tasmota and there is already a better way to do it.

But I think that when receiving messages is enabled (`TmState 3`), they should create a [Rule Trigger](https://tasmota.github.io/docs/Rules/#rule-trigger) upon reception of the message. And if the message is in the form `/command param1 param2 param3` it should be split so that we could use that sort of data with rules as well.

**Comment by ascillato** [comment, marked as answer]
Please, post the console log when you receive a telegram message.

**Reply by sfromis** [reply]
The general concept for triggers is that when you have a  JSON payload published to MQTT, you can have triggers to match the structure of the JSON, meaning that in the console, you can see what you have.

From what I see, it does not look like the Telegram integration is expected it to work via triggers, but via Tasmota commands being submitted from the Telegram client. Do read the thread about it. https://github.com/arendst/Tasmota/issues/8619

If you "need" triggers, you can issue Event commands or similar from the client.

**Reply by ascillato** [reply]
@sfromis is totally right. When you write something in Telegram, it is interpreted in Tasmota directly as a command in the console. So, if you want a trigger, you can use for example:

1- Setting a rule event In the Tasmota console as
```JS
Rule1 1
Rule1 ON Event#TurnAll DO power0 1 ENDON
```

2- Then, in Telegram you can call that event by: `EVENT TURNALL`

**Reply by mantielero** [reply]
For instance, if I send `TmState` via telegram I get the expected result:
```
07:45:23.125 RSL: RESULT = {"TmState":{"Send":"ON","Receive":"ON","Echo":"OFF"}}
```

Thanks for your suggestion. It is the way I will make it work, but IMHO it is not ideal. 

In telegram, bots has [commands](https://core.telegram.org/bots#commands). "Normal" bots would expect something like `/turnall` rather than `event turnall`.

Could it be possible at least to alias `/<something>` into `event <something>`?

In any case I wonder if having the telegram's payload directly available in the same way I get the payload from lets say a zigbee sensor, would enable other sorts of filtering. Such as filtering `chatid` for security.

Having available something like this:
```json
{"ok":true,"result":[
    {"update_id":973125394,
     "message":{"message_id":25,
                "from":{"id":139920293,"is_bot":false,"first_name":"Theo","last_name":"Arends","username":"tjatja","language_code":"nl"},
                "chat":{"id":139920293,"first_name":"Theo","last_name":"Arends","username":"tjatja","type":"private"},
                "date":1591877503,
                "text":"M1"
               }
    },
. . . 
}
```

I also wonder if in order to send a command to tasmota if it would be better to do something like:
```
/cmd <command>
```
rather than just the `<command>`.
